### PR TITLE
Panel classes for header and footer

### LINKED ISSUE #223: Panel footer without padding
**Describe the bug**
I use web components and therefore wrap elements for conditionals which breaks the panel footer because it uses a [child combinator](https://developer.mozilla.org/en-US/docs/Web/CSS/Child_combinator). 

```
<setup-display display-target="setup-file-upload" display-attribute="has-file">
 <footer>
  ...
 </footer>
</setup-display>
```

**To Reproduce**
- Create a panel with a footer 
- Wrap the footer with another element 

**Possible fix**
Add an optional class (`.panel-footer`).

We could just use a [descendant combinator](https://developer.mozilla.org/en-US/docs/Web/CSS/Descendant_combinator), but I prefer to find a general pattern for this scenario, even though multiple footers in the panel in quite unlikely.  

Maybe this applies to other components that use child combinator selectors ending with element such as `footer`?

**Desktop :**
 - OS: MAC
 - Browser: Firefox Developer Edition 
 - Version: 66

**Screenshots**
screenshot 2019-02-15 at 14 24 59

### PR DESCRIPTION
**Pull request by Akswii**
Fixes #223

### CODE REVIEW

File: src/App/Documentation/components/Panel/index.js [hunk 1]
```diff
@@ -59,17 +61,35 @@ const Overview = () => (
     <>
         <h2 id="overview">Overview</h2>
         <p>
-            A panel consists of three parts: A header, a body and a footer. You can mostly put whatever you want in any of these,
-            but try not to put too much in the header and footer as these are meant to contain short and concise information about the body content.
+            A panel consists of three parts: A header, a body and a footer.
+            You can mostly put whatever you want in any of these, but try not to put too much in the header and footer as these are meant to contain short and concise information about the body content.
```
Comment (hmosvoll) [inline src/App/Documentation/components/Panel/index.js]: 😄

File: src/App/Documentation/components/Panel/index.js [hunk 2]
```diff
@@ -59,17 +61,35 @@ const Overview = () => (
...
-                bodyContent={bodyContent}
-            />
+            >
+                {bodyContent}
+            </PanelComponent>
+
+        </ComponentPreview>
+        <p>
+            You can choose whether to use <Property value=".panel-header" /> and <Property value=".panel-footer" /> or not.
+            When using the <PrismCode className="language-html">{"<footer/>"}</PrismCode> or <PrismCode className="language-html">{"<header/>"}</PrismCode> they need to be direct children of <Property value=".panel" />.
```
Comment (hmosvoll) [inline src/App/Documentation/components/Panel/index.js]: What do we recommend? Most of our user do not have the knowledge to take these kind of decisions and is not experiencing the issue I had. Is it a possible solution move it to the bottom of the page and present it as an alternate markup when the DesignGuide user can't have the footer and header as a direct child?

File: src/App/Documentation/components/Panel/index.js (hunk 1 above)
Comment (Akswii) [reply src/App/Documentation/components/Panel/index.js]: ... any suggestions? 😆

File: src/App/Documentation/components/Panel/index.js (hunk 2 above)
Comment (Akswii) [reply src/App/Documentation/components/Panel/index.js]: Well, if we want to recommend the simplest markup then we need to explain this to them. I also feel like we should inform them about this possible issue as soon as possible so they don't spend time trying to figure out why their panel is not working like ours. Can we not expect our users to understand descendant selectors?

File: src/App/Documentation/components/Panel/index.js (hunk 1 above)
Comment (hmosvoll) [reply src/App/Documentation/components/Panel/index.js]: hehe, nope, I like the informal tone :)

File: src/App/Documentation/components/Panel/index.js (hunk 2 above)
Comment (hmosvoll) [reply src/App/Documentation/components/Panel/index.js]: But why should we in the Overview section explain something that does not apply to most users, and for them possibly create confusion. Is it many people experiencing this issue? Is this not an edge case?

> Can we not expect our users to understand descendant selectors?
A portion of them probably don't.

Another possibility is just to remove the doc for the footer and header without class, and only keep it for backwards compatibility.

Comment (Akswii) [reply src/App/Documentation/components/Panel/index.js]: Ok, does this mean using footers and headers without classes is deprecated? Or do we just do a "silent" change but keep the classless versions for backward compat.

Comment (hmosvoll) [reply src/App/Documentation/components/Panel/index.js]: I guess so.

File: src/App/Documentation/components/Panel/__snapshots__/index.test.js.snap [hunk 3]
```diff
@@ -16,19 +16,55 @@ exports[`Documentation: Panel Overview renders 1`] = `
...
+      </PrismCode>
+       is deprecated. Use our 
+      <Property
+        value=".panel-header"
+      />
+       and 
+      <Property
+        value=".panel-footer"
+      />
+       classes instead.
```
Comment (hmosvoll) [inline src/App/Documentation/components/Panel/__snapshots__/index.test.js.snap]: Details, but don't think "instead" is the best wording here, considering we don't remove anything.

File: CHANGELOG.md [hunk 4]
```diff
@@ -5,6 +5,7 @@
 ### Added
 
 - Added class .action-toggle to .action-list, selecting on .material-icons is deprecated.
+- Added class .panel-header and .panel-footer to panel. Using just <header> and <footer> is still valid.
```
Comment (hmosvoll) [inline CHANGELOG.md]: Considering the addition off a deprecation warning, maybe it should be more clearly reflected here?

**Pull request merged**